system/jlib/jtrace.cpp

```diff
@@ -819,9 +816,16 @@ class CServerSpan : public CSpan
 
 //---------------------------------------------------------------------------------------------------------------------
 
-std::chrono::nanoseconds nowTimesStampInNanos()
+const std::chrono::time_point<std::chrono::system_clock, std::chrono::nanoseconds> nowSystemClock()
```

Comment:
@rpastrana @ghalliday from a users perspective I was picturing something very simple and that could be used in the code as is.  There may be more powerful things you could do, and improvements could be made for example to support nanoseconds, but if we supported the following it would allow our existing code to work very easily:

 virtual ISpan * createServerSpan(const char * name, const IProperties * httpHeaders, const, unsigned msStartTIme, SpanFlags flags = SpanFlags::None) const = 0;
 
 Roxie would just pass in the existing variable "qstart" and it would be done.

Internally you would do something like "unsigned elapsed = msTick() - msStartTime;" and then use elapsed to adjust everything else.  I didn't look closely enough to see if something is wrong with that, but that's the idea.

ESP already has a similar value as well.

This prevents rearchitecting the servers now.  We can make further enhancements in the future.

Reply:
@afishbeck I did notice the ESP keeps a creation timestamp (in ms granularity) in the request context and the current design would likely force a change to track that timestamp in the granularity (nanos) used by OTel internals. 
Your suggestion seems ok w/ me if you're ok w/ the potential loss of precision. We'd have to confirm there are no other issues performing the duration casts between millis and a nanos.

Reply:
@afishbeck I think I jumped the gun on your suggestion. I assumed the context tracked an actual start timestamp, but it looks like it's more of a tick mark used to calculated elapsed time at completion time?

We're relying on OTel to calculate the span duration time, and only overriding the start time (in std::chrono::nanos) on-demand.
I think your suggestion is to offset that time stamp by a count of millis (elapsed) correct?

Reply:
right.  My suggestion was a way to make it easy to work with existing code.  The idea was to use the existing value to figure out the already elapsed time and use that as a delta to correct what Otel would currently generate.  But if we want to go ahead and refactor a bit to "do it right" that would be fine.  I guess if we don't do it now it may never happen.  There are places in both roxie and ESP now that get the current tick and later use that for reporting and other things.  We can change that how ever makes sense as long as current uses aren't broken and otels needs are met.

But I think we should decide what that roxie and ESP code will look like when deciding on the design for passing in the start time here.

Reply:
@ghalliday btw this is what I meant the other day when I said that changing surrounding code can help get the design right, but can also make the current PR bigger :)

Reply:
I'd have to do a little more research, but something like this **seems** to produce reasonable values (now - offsetMS)
systemClockTime = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::system_clock::now().time_since_epoch() - std::chrono::milliseconds(msTick() - msTicksOffset));

Anyway, have we considered migrating off of mstick in favor of a std::chrono based duration?
std::chrono::steady_clock::now() ?

Reply:
It wasn't on my radar, but definitely worth considering.